velox/exec/Spill.cpp

```diff
@@ -47,24 +68,27 @@ void SpillMergeStream::pop() {
 int32_t SpillMergeStream::compare(const MergeStream& other) const {
   VELOX_CHECK(!closed_);
   auto& otherStream = static_cast<const SpillMergeStream&>(other);
-  auto& children = rowVector_->children();
-  auto& otherChildren = otherStream.current().children();
-  int32_t key = 0;
-  if (sortCompareFlags().empty()) {
-    do {
-      auto result = children[key]
-                        ->compare(
-                            otherChildren[key].get(),
-                            index_,
-                            otherStream.index_,
-                            CompareFlags())
-                        .value();
-      if (result != 0) {
-        return result;
-      }
-    } while (++key < numSortKeys());
+  if (prefixComparatorEnabled_) {
+    VELOX_DCHECK(otherStream.prefixComparatorEnabled_);
+    VELOX_DCHECK_EQ(
+        sortLayout_.normalizedBufferSize,
+        otherStream.sortLayout_.normalizedBufferSize);
+    if (!sortLayout_.hasNonNormalizedKey) {
```

Comment:
`!(sortLayout_.hasNonNormalizedKey || sortLayout_.nonPrefixSortStartIndex < sortLayout_.numNormalizedKeys)`
Maybe introduce a method for this in `PrefixSortLayout`